finality-aleph/src/import.rs

```diff
@@ -54,15 +54,19 @@ where
         &mut self,
         block: BlockImportParams<Block, Self::Transaction>,
     ) -> Result<ImportResult, Self::Error> {
+        let number = *block.header.number();
         let post_hash = block.post_hash();
         self.metrics
             .report_block(post_hash, Instant::now(), Checkpoint::Importing);
 
         let result = self.inner.import_block(block).await;
 
-        if let Ok(ImportResult::Imported(_)) = &result {
+        if let Ok(ImportResult::Imported(ref aux)) = &result {
             self.metrics
                 .report_block(post_hash, Instant::now(), Checkpoint::Imported);
+            if aux.is_new_best {
```

Comment:
At first glance, it could be the case that we are importing blockA which becomes the best, then we import blockB which also becomes the best, but because of the concurrency they execute in different order in line 67, resulting in blockA being reported as best. If I'm not mistaken, this shouldn't be a problem for us, because we are never importing two blocks in one branch at the same time, and when importing two blocks from different branches we should quickly reject one branch.

Reply:
Changed to use best block from client, which is uglier, but should be safe.